### PR TITLE
Include observers and offline players when showing match stats

### PR DESCRIPTION
**Pull request by KingOfSquares**
Also re-orders and improves some logic
Signed-off-by: KingSimon <19822231+KingOfSquares@users.noreply.github.com>

### CODE REVIEW

File: core/src/main/java/tc/oc/pgm/db/CacheDatastore.java [hunk 1]
```diff
@@ -59,6 +70,16 @@ public Settings getSettings(UUID id) {
...
+  @Override
+  public void setSkin(UUID uuid, Skin skin) {
+    skins.put(uuid, skin);
+  }
+
+  @Override
+  public Skin getSkin(UUID id) {
+    return skins.getUnchecked(id);
+  }
+
```
Comment (KingOfSquares) [inline core/src/main/java/tc/oc/pgm/db/CacheDatastore.java:82]: Should this store a `DatabaseSkin` object to be similar to the other stuff?

**Review by Pablete1234 — APPROVED** [review]

File: core/src/main/java/tc/oc/pgm/api/Datastore.java [hunk 2]
```diff
@@ -24,6 +25,22 @@
...
+   */
+  void setSkin(UUID uuid, Skin skin);
+
+  /**
+   * Get the skin for a given player {@link UUID}.
+   *
+   * @param uuid The {@link UUID} of a player
+   * @return A {@link Skin}
+   */
+  Skin getSkin(UUID uuid);
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/api/Datastore.java:42]: Skins will explode the size of the database.. are we sure?

Comment (Electroid) [reply core/src/main/java/tc/oc/pgm/api/Datastore.java:42]: Oh, I just realized it's only implemented by cache, ok.

Comment (KingOfSquares) [reply core/src/main/java/tc/oc/pgm/api/Datastore.java:42]: Yes, my thought as well :+1:

**Pull request merged**